internal/controller/cilium_flow_manager.go

```diff
@@ -0,0 +1,228 @@
+				},
+				ProxyPort:        int32(flowObj.GetProxyPort()),
+				EgressAllowedBy:  convertCiliumPolicies(flowObj.GetEgressAllowedBy()),
+				IngressAllowedBy: convertCiliumPolicies(flowObj.GetIngressAllowedBy()),
+				EgressDeniedBy:   convertCiliumPolicies(flowObj.GetEgressDeniedBy()),
+				IngressDeniedBy:  convertCiliumPolicies(flowObj.GetIngressDeniedBy()),
+			}
+
+			err = sendNetworkFlowsData(&sm, &ciliumFlow)
+			if err != nil {
+				fm.logger.Errorw("Cannot send object metadata", "error", err)
+				return err
+			}
+		}
+	}
+}
+
+// convertLayer4 function converts a slice of flow.Layer4 objects to a slice of pb.Layer4 objects.
+func convertLayer4(l4 *flow.Layer4) *pb.Layer4 {
+	layer4 := &pb.Layer4{}
+	if l4 == nil {
+		return layer4
+	}
+	switch protocol := l4.Protocol.(type) {
+	case *flow.Layer4_TCP:
+		layer4.Protocol = &pb.Layer4_Tcp{
+			Tcp: &pb.TCP{
+				SourcePort:      protocol.TCP.SourcePort,
+				DestinationPort: protocol.TCP.DestinationPort,
+			},
+		}
+	case *flow.Layer4_UDP:
+		layer4.Protocol = &pb.Layer4_Udp{
+			Udp: &pb.UDP{
+				SourcePort:      protocol.UDP.SourcePort,
+				DestinationPort: protocol.UDP.DestinationPort,
+			},
+		}
+	case *flow.Layer4_ICMPv4:
+		layer4.Protocol = &pb.Layer4_Icmpv4{
+			Icmpv4: &pb.ICMPv4{
+				Type: protocol.ICMPv4.Type,
+				Code: protocol.ICMPv4.Code,
+			},
+		}
+	case *flow.Layer4_ICMPv6:
+		layer4.Protocol = &pb.Layer4_Icmpv6{
+			Icmpv6: &pb.ICMPv6{
+				Type: protocol.ICMPv6.Type,
+				Code: protocol.ICMPv6.Code,
+			},
+		}
+	case *flow.Layer4_SCTP:
+		layer4.Protocol = &pb.Layer4_Sctp{
+			Sctp: &pb.SCTP{
+				SourcePort:      protocol.SCTP.SourcePort,
+				DestinationPort: protocol.SCTP.DestinationPort,
+			},
+		}
+	default:
+	}
+
+	return layer4
+}
+
+// convertCiliumWorkflows function converts a slice of flow.Workload objects to a slice of pb.Workload objects.
+func convertCiliumWorkflows(workloads []*flow.Workload) []*pb.Workload {
+	protoWorkloads := []*pb.Workload{}
+	for _, workload := range workloads {
+		protoWorkload := &pb.Workload{
+			Name: workload.GetName(),
+			Kind: workload.GetKind(),
+		}
+		protoWorkloads = append(protoWorkloads, protoWorkload)
+	}
+	return protoWorkloads
+}
+
+// convertCiliumPolicies function converts a slice of flow.Policy objects to a slice of pb.Policy objects.
+func convertCiliumPolicies(policies []*flow.Policy) []*pb.Policy {
+	protoPolicies := []*pb.Policy{}
+	for _, policy := range policies {
+		protoPolicy := &pb.Policy{
+			Name:      policy.GetName(),
+			Namespace: policy.GetNamespace(),
+			Labels:    policy.GetLabels(),
+			Revision:  policy.GetRevision(),
+		}
+		protoPolicies = append(protoPolicies, protoPolicy)
+	}
+	return protoPolicies
+}
+
+// readFlows uses the observerClient to make gRPC calls to hubble-relay and grab the last x amount of flows.
+func (fm *FlowManager) readFlows(ctx context.Context) ([]*observer.GetFlowsResponse, error) {
+	req := &observer.GetFlowsRequest{
+		Number: 10,
```

Comment:
Make this configurable. That's a performance tuning knob which I'm not sure how to set right now.
At the very minimum, define this as a constant.